dali/pipeline/data/tensor_list_test.cc

```diff
@@ -504,26 +483,24 @@ TYPED_TEST(TensorListTest, TestShareData) {
   // Check the internals
   ASSERT_TRUE(tensor_list2.shares_data());
   ASSERT_EQ(tensor_list2.raw_data(), tensor_list.raw_data());
-  ASSERT_EQ(tensor_list2.nbytes(), tensor_list.nbytes() / sizeof(float) * sizeof(uint8));
+  ASSERT_EQ(tensor_list2.nbytes(), tensor_list.nbytes());
   ASSERT_EQ(tensor_list2.ntensor(), tensor_list.ntensor());
   ASSERT_EQ(tensor_list2.size(), tensor_list.size());
   for (size_t i = 0; i < tensor_list.ntensor(); ++i) {
     ASSERT_EQ(tensor_list2.tensor_shape(i), shape[i]);
     ASSERT_EQ(tensor_list2.tensor_offset(i), offsets[i]);
   }
 
-  // Trigger allocation through buffer API, verify we no longer share
-  tensor_list2.template mutable_data<double>();
+  // Trigger allocation through buffer API, verify we cannot do that
+  ASSERT_THROW(tensor_list2.template mutable_data<double>(), std::runtime_error);
+  tensor_list2.Reset();
   ASSERT_FALSE(tensor_list2.shares_data());
 
   // Check the internals
-  ASSERT_EQ(tensor_list2.size(), tensor_list.size());
-  ASSERT_EQ(tensor_list2.nbytes(), tensor_list.nbytes() / sizeof(float) * sizeof(double));
-  ASSERT_EQ(tensor_list2.ntensor(), tensor_list.ntensor());
-  for (size_t i = 0; i < tensor_list.ntensor(); ++i) {
-    ASSERT_EQ(tensor_list2.tensor_shape(i), shape[i]);
-    ASSERT_EQ(tensor_list2.tensor_offset(i), offsets[i]);
-  }
+  ASSERT_EQ(tensor_list2.size(), 0);
```

Comment:
A general remark: `EXPECT_` is preferable if the test can safely continue despite failure.